prometheus/client.go

```diff
@@ -65,7 +65,16 @@ func (in *Client) GetSourceServices(namespace string, servicename string) (map[s
 			}
 			source := fmt.Sprintf("%s/%s", sourceService, metric["source_version"])
 			if arr, ok := routes[index]; ok {
-				routes[index] = append(arr, source)
+				found := false
+				for _, s := range arr {
+					if s == source {
+						found = true
```

Comment:
That's where I find golang is a bit too rough. There's no nice API to deal with slices.

Reply:
I was wondering if there was some "magic" inbuilt function that I didn't find it, so I used the old school style here :-).